lib/ansible/module_utils/basic.py

```diff
@@ -1629,12 +1629,17 @@ def _check_type_str(self, value):
             return check_type_str(value, allow_conversion)
         except TypeError:
             common_msg = 'quote the entire value to ensure it does not change.'
+            from_msg = '{0!r}'.format(value)
+            to_msg = '{0!r}'.format(to_text(value))
+            if param is not None:
+                from_msg = '{0}: {1!r}'.format(param, value)
+                to_msg = '{0}: {1!r}'.format(param, to_text(value))
             if self._string_conversion_action == 'error':
                 msg = common_msg.capitalize()
                 raise TypeError(to_native(msg))
             elif self._string_conversion_action == 'warn':
-                msg = ('The value {0!r} (type {0.__class__.__name__}) in a string field was converted to {1!r} (type string). '
-                       'If this does not look like what you expect, {2}').format(value, to_text(value), common_msg)
+                msg = ("The value '{0}' (type {1.__class__.__name__}) was converted to '{2}' (type string). "
```

Comment:
I'd recommend changing `'{0}'` and `'{2}'` to use double quotes instead of single.  `repr` usually gives single quotes, and as such you end up with something like:

```
was converted to 'password: '********'' (type string)
```

Reply:
Good point. Thanks!